pkg/stack/tile_stack.go

```diff
@@ -0,0 +1,77 @@
+package stack
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+type Stack[T interface{}] struct {
+	seed    int64
+	turn_no int32
+	tiles   []T
+	order   []int32
+}
+
+type StackOutOfBoundsError struct{}
+
+func (e *StackOutOfBoundsError) Error() string {
+	return fmt.Sprint("stack: out of bounds")
+}
```

Comment:
nit: this could be simplified to:
```suggestion
var StackOutOfBoundsError = errors.New("stack: out of bounds")
```
This is my bad on the terminology since I explicitly asked for a separate error type when I just wanted something that can be easily checked against with i.e. `errors.Is()`